src/04_Make_country_level_predictions.R

```diff
@@ -0,0 +1,825 @@
+desired_version <- "2.0.3"
+
+# Check if caretEnsemble is installed
+if ("caretEnsemble" %in% rownames(installed.packages())) {
+  # Get the current version of caretEnsemble
+  current_version <- packageVersion("caretEnsemble")
+  # Compare current version with the desired version
+  if (as.character(current_version) != desired_version) {
+    # Uninstall the current version if it's not the desired version
+    remove.packages("caretEnsemble")
+    # Install the specific version
+    devtools::install_github("zachmayer/caretEnsemble@2.0.3")
+    # Load 
+    library(caretEnsemble)
+  } else {
+    library(caretEnsemble)
+  }
+  
+} else {
+  # If caretEnsemble is not installed, install the specific version
+  devtools::install_github("zachmayer/caretEnsemble@2.0.3")
+  # Load 
+  library(caretEnsemble)
+  rm(current_version, desired_version)
+}
+
+
+#--------------------------------------------
+#----------  To do: specify country  --------
+#--------------------------------------------
+#If you'd like to predict for another country, change the shapefile
+country_name<-"Belgium"
+country<-sf::st_read(here("./data/external/GIS/Belgium/belgium_boundary.shp"))
+country_ext<-terra::ext(country) 
+country_vector <- terra::vect(country) #Convert to a SpatVector, used for masking
+
+
+#--------------------------------------------
+#------------ Load country shape ------------
+#--------------------------------------------
+#Only used when no EU model could be fitted
+belgium<-rnaturalearth::ne_countries(country="Belgium", scale=10)[1]
+belgium_ext<-terra::ext(belgium) 
+belgium_vector <- terra::vect(belgium) #Convert to a SpatVector, used for masking
+
+
+#--------------------------------------------
+#----------- Load taxa info  ----------------
+#--------------------------------------------
+taxa_info<-read.csv2(paste0("./data/projects/",projectname,"/",projectname,"_taxa_info.csv"))
+accepted_taxonkeys<-taxa_info%>%
+  pull(speciesKey)%>%
+  unique()
+
+
+#-------------------------------------------------
+#---------- Load habitat raster data -------------
+#-------------------------------------------------
+#ONE LAYER HAS A SLIGHTLY DIFFERENT EXTENT: CUT ALL OTHERS TO THIS EXTENT
+habitat<-list.files((here("./data/external/habitat")),pattern='tif',full.names = T)
+habitat_stack<-rast(habitat[c(1:5,7)]) #Distance to water (layer 6) has another extent and we're not sure whether it is correct: leave it out!
+#habitat_stack2<-rast(habitat[6])
+
+#habitat_stack1<-crop(habitat_stack1, ext(habitat_stack2))
+#habitat_stack<-c(habitat_stack1, habitat_stack2)
+
+
+#-------------------------------------------------
+#---------- Define resampling raster -------------
+#-------------------------------------------------
+#Used as a template when global model predictions need to be in same raster format as those for the EU model
+resampling_raster<-rast(here("./data/external/climate/byEEA_finalRCP/belgium_rcps/rcp26/anngdd100.tif"))
```

Comment:
This file seems to be missing from the setup. 

```
Error: [rast] file does not exist: C:/Users/sander_devisscher/Documents/GitHub/risk-modelling-and-mapping/./data/external/climate/byEEA_finalRCP/belgium_rcps/rcp26/anngdd100.tif
In addition: Warning message:
C:/Users/sander_devisscher/Documents/GitHub/risk-modelling-and-mapping/./data/external/climate/byEEA_finalRCP/belgium_rcps/rcp26/anngdd100.tif: No such file or directory (GDAL error 4) 
```

Reply:
2 issues "./" does some weird shit with `here()` and `anngdd100.tif == var10_30yrMeanAnnualCumulatedGDDAbove5degreesC_rcp26_2041_2070_Belgium.tif`. At least that how I understand it.

Reply:
`here::here("data", "external", "climate", "byEEA_finalRCP", "rcp26", "anngdd100.tif")`